src/main/java/co/inventorsoft/academy/collections/model/Range.java

```diff
@@ -3,58 +3,106 @@
 import java.util.Collection;
 import java.util.Iterator;
 import java.util.Set;
+import java.util.function.Function;
+
 
 public class Range<T> implements Set<T> {
+    private Set<T> mySetlist = new java.util.HashSet<>();
```

Comment:
Using set is forbidden, it's cheating.

Reply:
It should be based on collection framework’s Set interface.- no, it isnt)

Reply:
I don't get

Reply:
in the condition of the problem it is said -"It should be based on collection framework’s Set interface.". So it meen that i can use set . And it mean it is not cheating.

Reply:
Using Set for building a Set is cheating because it loses the idea of learning how Set works and how to achieve functionality which is required. Your Range solution just delegates methods to underlying Set.

By "based on collection framework's Set interface" means to implement Set interface.

Also, when homework was given I mentioned specifically not to use Set or Map as the solution under the hood.